### ISSUE TITLE
Update 6.2.0 for PHP 8 

### ISSUE THREAD

**Issue by Jimmi08** [Jimako-e107-plugins/lgsl#2]
PHP Fatal error:  Uncaught ValueError: max(): Argument #1 ($value) must contain at least one element \userbar.php:36

**Comment by LaocheXe** [comment]
Question: userbar.php just makes the images/banners, for the servers correct? Maybe add support for webp - who knows. I'm willing to throw myself at this a little - tried my own game server list - only difference is mine is hard coded in, no admin area - ear server listed has like three files, two php pages, and one js file to auto-refresh cause I cant get the js to just refresh the page - https://501stlegion-a3.com/server_ten.php

**Comment by Jimmi08** [comment]
Hi, in fact I am not sure, I am just fixing stuff, you know. 
Something like this (It is hard to test it on localhost):
![image](https://github.com/Jimako-e107-plugins/lgsl/assets/5429548/7546b3b3-890f-44d9-9724-8ea3e638195d)

**Comment by LaocheXe** [comment]
Right, so having some game servers can help then. Going to set some up on my testing site.

**Comment by Jimmi08** [comment]
There is lgsl 7 on the way, so I didn't want to change files too much. But if there is need, more of stuff can be replaced by e107  (e107::getDb()) and so. Only it will complicate next update, so I haven't touched it.

**Comment by LaocheXe** [comment]
True, I mean we can tinker a little with it, and see what LGSL 7 looks like - https://testing.501stlegion-a3.com/LGSL/

So Index.php needs a look at, have links that return nothing on my end, For example when I click on the TeamSpeak 3 server i get a url - index.php?s=ts3.501stlegion-a3.c%E2%80%8Bom%E2%80%8B  <-- idk if it calls for userbar or not on the index so not sure - the other teamspeak 3 link works because it connects you to the server - all them work but the arma 3 ones, the servers use mods and they would need to be added to the link kind of like my Voice eXe plugin with the custom links but thats something else to deal with.

**Comment by Jimmi08** [comment]
You have there fatal error Fatal error: Uncaught Error: Call to a member function retrieve() on null in /usr/www/taskforce13tsw/501st/e107-test/e107_plugins/calendar_menu/calendar_menu.php

I need your preferences - just export them from export tools. 

My test data looks like this:
![image](https://github.com/Jimako-e107-plugins/lgsl/assets/5429548/0100f7df-c5ee-453e-a43d-4e3002a0724a)
My URLs looks correct, your are ugly. Maybe spaces in names?

![image](https://github.com/Jimako-e107-plugins/lgsl/assets/5429548/22941755-14df-4572-9167-a406c4ddf5f4)

**Comment by LaocheXe** [comment]
That fatal error is for out-dated event calendar menu (NFOServers controls the php version, and forced it to 8.something) - was on php 7.4 for the longest.

I'll come around to fix that later on. As for the space, I didn't give the names, just added the IP/Host, Connection Port, and Query Port - I don't know what software port is.

[lgsl_e107Export_2024-01-01.zip](https://github.com/Jimako-e107-plugins/lgsl/files/13806304/lgsl_e107Export_2024-01-01.zip)

**Comment by Jimmi08** [comment]
You should first install lgsl as a standalone version, then check the e107 version - to see where the is problem. 
Version 6 is not for PHP 8 either if I know. 
Thanks for data.
By the way - that fatal error can influence this plugin too.  If you use my version of the calendar plugin, report it and I will try to fix it.
